lib/xverifier/applicator_with_options.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+module XVerifier
+  # An applicator with usefull options
+  # @example if
+  #   ApplicatorWithOptions.new(action, if: true)
+  # @example unless
+  #   ApplicatorWithOptions.new(action, unless: false)
+  # @attr action [Applicator]
+  #   main action to applicate on call
+  # @attr if_condition [Applicator]
+  #   main action only applicate if this applicates to truthy value
```

Comment:
evaluates is more generic and does not describe, that this stuff would be applicated